packages/core/test/predictableExec.test.ts

```diff
@@ -0,0 +1,587 @@
+    service.send({ type: 'NEXT' });
+
+    expect(eventArg.type).toBe('RAISED');
+  });
+
+  it('should call raised transition builtin actions with source event', () => {
+    let eventArg: any;
+    const machine = createMachine({
+      predictableActionArguments: true,
+      context: {},
+      initial: 'a',
+      states: {
+        a: {
+          on: {
+            NEXT: 'b'
+          }
+        },
+        b: {
+          on: {
+            RAISED: {
+              target: 'c',
+              actions: assign((_ctx, ev) => {
+                eventArg = ev;
+                return {};
+              })
+            }
+          },
+          entry: raise('RAISED')
+        },
+        c: {}
+      }
+    });
+
+    const service = interpret(machine).start();
+    service.send({ type: 'NEXT' });
+
+    expect(eventArg.type).toBe('RAISED');
+  });
+
+  it('should call invoke creator with source event', () => {
+    let eventArg: any;
+    const machine = createMachine({
+      predictableActionArguments: true,
+      context: {},
+      initial: 'a',
+      states: {
+        a: {
+          on: {
+            NEXT: 'b'
+          }
+        },
+        b: {
+          on: {
+            RAISED: 'c'
+          },
+          entry: raise('RAISED')
+        },
+        c: {
+          invoke: {
+            src: (_ctx, ev) => {
+              eventArg = ev;
+              return () => {};
+            }
+          }
+        }
+      }
+    });
+
+    const service = interpret(machine).start();
+    service.send({ type: 'NEXT' });
+
+    expect(eventArg.type).toBe('RAISED');
+  });
+
+  it('invoked child should be available on the new state', () => {
+    const machine = createMachine({
+      predictableActionArguments: true,
+      context: {},
+      initial: 'a',
+      states: {
+        a: {
+          on: {
+            NEXT: 'b'
+          }
+        },
+        b: {
+          invoke: {
+            id: 'myChild',
+            src: () => () => {}
+          }
+        }
+      }
+    });
+
+    const service = interpret(machine).start();
+    service.send({ type: 'NEXT' });
+
+    expect(service.state.children.myChild).toBeDefined();
+  });
+
+  it('invoked child should not be available on the state after leaving invoking state', () => {
+    const machine = createMachine({
+      predictableActionArguments: true,
+      context: {},
+      initial: 'a',
+      states: {
+        a: {
+          on: {
+            NEXT: 'b'
+          }
+        },
+        b: {
+          invoke: {
+            id: 'myChild',
+            src: () => () => {}
+          },
+          on: {
+            NEXT: 'c'
+          }
+        },
+        c: {}
+      }
+    });
+
+    const service = interpret(machine).start();
+    service.send({ type: 'NEXT' });
+    service.send({ type: 'NEXT' });
+
+    expect(service.state.children.myChild).not.toBeDefined();
+  });
+
+  it('should automatically enable predictableActionArguments', () => {
```

Comment:
If I'm understanding this correctly...

```suggestion
  it('should automatically enable preserveActionOrder', () => {
```